### PR TITLE
Docs: All Static Builds

### PR DESCRIPTION
**Pull request by ax3l**
Restructure the build options and give hints on all-static builds.

### CODE REVIEW

File: docs/source/dev/buildoptions.rst [hunk 1]
```diff
@@ -21,7 +24,41 @@ CMake Option                   Values          Description
...
 :sup:`1` *not yet implemented*
-:sup:`2` e.g. C++ keywords, currently disabled only for MSVC*
+
+:sup:`2` e.g. C++ keywords, currently disabled only for MSVC
+
+
+Shared or Static
+----------------
+
+By default, we will build as a static library (``libopenPMD.a`` or ``openPMD.lib``) and installs also its headers.
```
Comment (anokfireball) [inline docs/source/dev/buildoptions.rst]: Typo: installs -> install.

File: docs/source/dev/buildoptions.rst [hunk 2]
```diff
@@ -21,7 +24,41 @@ CMake Option                   Values          Description
...
-:sup:`2` e.g. C++ keywords, currently disabled only for MSVC*
+
+:sup:`2` e.g. C++ keywords, currently disabled only for MSVC
+
+
+Shared or Static
+----------------
+
+By default, we will build as a static library (``libopenPMD.a`` or ``openPMD.lib``) and install also its headers.
+You can only build a static or a shared library (``libopenPMD.so`` or ``openPMD.dll``) at a time.
```
Comment (anokfireball) [inline docs/source/dev/buildoptions.rst]: I had to read this a few times to understand what you mean. There are really two sentences here - one about either static or shared - and one about the names of the shared library.

Comment (ax3l) [reply docs/source/dev/buildoptions.rst]: I can put the brackets of the first sentence in the second one.

Comment (anokfireball) [reply docs/source/dev/buildoptions.rst]: Would make it easier to understand IMO.

**Pull request merged**